pyramid_oereb/views/webservice.py

```diff
@@ -199,55 +233,68 @@ def get_extract_by_id(self):
-            elif params.format == 'pdf':
-                log.debug("get_extract_by_id() calling pdf")
-                response = render_to_response(
-                    'pyramid_oereb_extract_print',
-                    (extract, params),
-                    request=self._request
+            else:
+                raise HTTPBadRequest("Missing required argument")
+            # check if result is strictly one (we queried with primary keys)
+            if len(real_estate_records) == 1:
+                extract = processor.process(
+                    real_estate_records[0],
+                    params,
+                    self._request.route_url('{0}/sld'.format(route_prefix))
                 )
+                if params.format == 'json':
+                    log.debug("get_extract_by_id() calling json")
+                    response = render_to_response(
+                        'pyramid_oereb_extract_json',
+                        (extract, params),
+                        request=self._request
+                    )
+                elif params.format == 'xml':
+                    log.debug("get_extract_by_id() calling xml")
+                    response = render_to_response(
+                        'pyramid_oereb_extract_xml',
+                        (extract, params),
+                        request=self._request
+                    )
+                elif params.format == 'pdf':
+                    log.debug("get_extract_by_id() calling pdf")
+                    response = render_to_response(
+                        'pyramid_oereb_extract_print',
+                        (extract, params),
+                        request=self._request
+                    )
+                else:
+                    raise HTTPBadRequest("The format '{}' is wrong".format(params.format))
+                end_time = timer()
+                log.debug("DONE with extract, time spent: {} seconds".format(end_time - start_time))
             else:
-                raise HTTPBadRequest("The format '{}' is wrong".format(params.format))
-            end_time = timer()
-            log.debug("DONE with extract, time spent: {} seconds".format(end_time - start_time))
-            return response
-        else:
-            return HTTPNoContent("No real estate found")
+                raise HTTPNoContent("No real estate found")
+        except HTTPNoContent as err:
+            response = HTTPNoContent('{}'.format(err))
+        except HTTPBadRequest as err:
+            response = HTTPBadRequest('{}'.format(err))
+        try:
+            response.extras = OerebStats(service='GetExtractById',
+                                         output_format=params.format,
+                                         params=vars(params))
+        except Exception as err:
+            # if params is not set we get UnboundLocalError
+            # or we could get ValueError
+            # in any case, the logging should never crash the response deliver
```

Comment:
the response delivery

Reply:
now we get the error message from the original error in the logs:
https://github.com/danduk82/pyramid_oereb/pull/2/commits/39b9c20f9ed19f48e64874f7fb0fb1ad88c036ec#diff-d3c05cc6b24ba1ebfd0188a6b955c10bR289